src/gt4sd/algorithms/core.py

```diff
@@ -700,6 +790,108 @@ def ensure_artifacts(self) -> str:
         return self.ensure_artifacts_for_version(self.algorithm_version)
 
 
+class ConfigurablePropertyAlgorithmConfiguration(AlgorithmConfiguration):
```

Comment:
wondering why all the type ignore in the methods, what is the error reported?

Reply:
It's because of the parent class behavior in all cases

- So the ones in the contstructor for lines like `self.domain=domain` says: `error: Cannot assign to class variable "domain" via instance`. That's because in the parent class (`AlgorithmConfiguration`) we set it as `domain: ClassVar[str]` so its 
- the ones in the signatures like `get_application_prefix` which returns a `str` are because in the parent class those are class methods, not instance methods. THe error is `Signature of "get_application_prefix" incompatible with supertype "AlgorithmConfiguration`

It might be fixable by a refactor but I'm not sure it's worth it

Reply:
Makes sense, can you track this in an issue with the improvement tag? So that we keep it on the radar.